### PR TITLE
fix(diffview equalise): avoid resize on diff(view) and fix equalise.

### PR DESCRIPTION
**Pull request by ianbattersby**
I've captured my changes so far in this PR for interest/comment. I don't particularly comfortable about the `autocmd` but it's the only thing that works for me at the moment. I'm pretty confident that [diffview](https://github.com/sindrets/diffview.nvim) is problematic in setting file and buffer types after creation too, more than I've seen with other plugins that have to use `vim.schedule` to circumvent this. Even [my theme](https://github.comful1e5/onedark.nvim) (that flawlessly darkens certain panels based on filetype) fails to work for `DiffviewFiles` 😬

Changes:

* `:DiffviewOpen` would open _without_ the panes equalised with the navigation _always_ taking the larger potion (`winwidth` seemingly defaulting to 84 for new file buffers). The subsequent call in `resized.lua` to `:FocusEqualise` (or indeed calling it manually) didn't correct it and honoured the 84. The only way to make it behave was to call `:set winwidth=20` _before_ calling `:DiffviewOpen`. So, I ended up adding an `autocmd` :(

* Noticed automatic the resizing behaviour didn't seem to honour the equalisation or exclusion. The last diff pane would marginally resize upon entering/leaving. I soon learnt that diff panes can have an empty buffer type and thus not fall back to the configured `nofile` exclusion. It seemed futile to do anything on these panes as one could argue `:FocusEqualise` must have been called by this point, so  I've used `vim.api.nvim_win_get_option(0, 'diff') ` to detect this state and return early.

### CODE REVIEW

File: lua/focus/modules/autocmd.lua [hunk 1]
```diff
@@ -39,6 +39,7 @@ function M.setup(config)
 			-- Which means focus will initially read it as blank buffer and resize. This is an issue for many other plugins that read ft too.
 			{ 'WinEnter,BufEnter', '*', 'lua vim.schedule(function() require"focus".resize() end)' },
 			{ 'WinEnter,BufEnter', 'NvimTree', 'lua require"focus".resize()' },
+			{ 'WinEnter', '*', 'set winwidth=20' },
```
Comment (ianbattersby) [inline lua/focus/modules/autocmd.lua:42]: I'd really welcome some alternative suggestions to this, feel like a hammer 😢

File: lua/focus/modules/resizer.lua [hunk 2]
```diff
@@ -26,15 +26,19 @@ end
 -- TODO: Support: Vista.vim
 -- TEST: floating windows, snap/telescope, toggleterm, trees, scrollview.nvim, blank buffer, popups during autocompletion i.e coq
 function M.split_resizer(config) --> Only resize normal buffers, set qf to 10 always
+	if vim.g.enabled_focus_resizing == 0 or vim.api.nvim_win_get_option(0, 'diff') then
```
Comment (ianbattersby) [inline lua/focus/modules/resizer.lua:29]: I haven't tested this with other differing scenarios yet and may need to be `vim.api.nvim_win_get_option(0, 'diff') and ft =~ nil` clause instead 🤔

**Comment by beauwilliams** [comment]
I'm going to try and find some time this evening to review. I totally understand your pain. I found similar issues trying to support some 3rd party plugins as there is no stock standard way to manage buffer creation at the moment

**Review by beauwilliams — APPROVED** [review]

**Comment by beauwilliams** [comment]
Thanks @ianbattersby 
I will merge this in now with the changes. I think this is the best solution for diffview for the time being. It does prevent that last window to the right resizing.

**Pull request merged**